catroid/src/main/java/org/catrobat/catroid/content/actions/EditLookAction.kt

```diff
@@ -66,32 +66,61 @@ class EditLookAction : PocketPaintAction() {
         }
     }
 
+    override fun restart() {
+        val stageActivity = StageActivity.activeStageActivity.get()
+        if (stageActivity != null && scope?.sprite?.isClone == true) {
+            protectOriginalLookData()
+        }
+        super.restart()
+    }
+
     override fun onIntentResult(resultCode: Int, data: Intent?) {
         val stageActivity = StageActivity.activeStageActivity.get()
         if (resultCode == Activity.RESULT_OK && stageActivity != null) {
             setLookData()
         }
         LookRequester.anyAsked = false
         responseReceived = true
-        stageActivity?.onResume()
+    }
+
+    @VisibleForTesting
+    fun protectOriginalLookData() {
+        val sprite = scope?.sprite ?: return
+        val oldLookData = sprite.look?.lookData ?: return
+        val lookDataName = oldLookData.name ?: return
+        val oldLookDataFile = sprite.look?.lookData?.file ?: return
+
+        try {
+            val newLookDataFile = StorageOperations.duplicateFile(oldLookDataFile)
```

Comment:
Those files aren't cleaned up when the stage is finished. There is the `LookFileGarbageCollector` but it is (yet) only run when the project is loaded. So when I execute the `EditLookAction` with clones a couple of times and then export the project to disk all those temporary look files will also end up in the .catrobat-archive.